### ISSUE TITLE
Filter out ignored dependencies before fetching

### ISSUE THREAD

**Issue by disposedtrolley** [dependabot/dependabot-core#1959]
**Package manage/ecosystem**
gomod
**Manifest contents prior to update**
N/A
**Updated dependency**
N/A
**What you expected to see, versus what you actually saw**
Attempting to `ignore` private Go modules using a wildcard name, e.g.

```
ignore:
      - dependency-name: "*github.com/myorg*"
```

still results in an error when Dependabot attempts to fetch the entire dependency list -- `Dependabot can't resolve your Go dependency files.`

The [docs](https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#specifying-dependencies-and-versions-to-ignore) note that `GitHub Dependabot checks for all allowed dependencies and then filters out any ignored dependencies or versions.` which explains why the fetching fails when it hits a private dependency midway through.

Would it make sense to filter out ignored dependencies prior to performing the fetch?

**Images of the diff or a link to the PR, issue or logs**
N/A

**Comment by Okeanos** [comment]
Please re-open this issue. Just because a workaround exists for some use cases doesn't mean this is actually solved. In particular please check my comment in #1884 . Having Dependabot fetch ignored dependencies before discarding them causes it to fail without being able to recover reproducibly and unnecessarily.

**Comment by jeffwidman** [comment]
@owenhaynes please file a new issue, this is related to filtering ignored deps, not GOPROXY.

Speaking of filtering ignored deps before fetch, I looked at this a little more, and I think this isn't something we will do anytime for the forseeable future.

As @feelepxyz [said above](https://github.com/dependabot/dependabot-core/issues/1959#issuecomment-646546368):
>  Dependabot will exclude the ignored dependency when checking for new versions but as soon as it attempts to update any other dependency it will try to resolve all dependencies in your manifest/lockfile. This is usually done by the package manager and we don't have much control over it. We have some ideas to improve this by stubbing out packages but going to be a while before we can look at this.

I don't think we'll very likely to go the route of stubbing out packages. We'd rather move Dependabot toward being the glue code between the native package manager and your repo, where Dependabot handles the cron, wiring up the native package manager, and then opening the PR. But move away from "re-implementation of native package manager behavior in Ruby, but poorly"... So I just don't see us doing pre-filtering in a way that doesn't force us to be pretty invasive in the native package manager... post-filtering is much easier.

I realize that means some deps will continue to break things, and I wish there were a better answer. Some package managers allow substitution or replacing deps within their manifest files, so it is something that can be supported at the package manager level.
